src/udev/udevd.c

```diff
@@ -1745,16 +1747,18 @@ static int run(int argc, char *argv[]) {
                 return r;
 
         if (arg_children_max == 0) {
+                unsigned long cpu_limit, mem_limit;
+                unsigned long cpu_count = 1;
                 cpu_set_t cpu_set;
-                unsigned long mem_limit;
-
-                arg_children_max = 8;
 
                 if (sched_getaffinity(0, sizeof(cpu_set), &cpu_set) == 0)
-                        arg_children_max += CPU_COUNT(&cpu_set) * 8;
+                        cpu_count = CPU_COUNT(&cpu_set);
+
+                cpu_limit = cpu_count * 2 + 16;
+                mem_limit = MAX(physical_memory() / (128UL*1024*1024), 10U);
 
-                mem_limit = physical_memory() / (128LU*1024*1024);
-                arg_children_max = MAX(10U, MIN(arg_children_max, mem_limit));
+                arg_children_max = MIN(cpu_limit, mem_limit);
+                arg_children_max = MIN(WORKER_NUM_MAX, arg_children_max);
```

Comment:
Yeah, something like that sounds reasonable.
But I think this would be much clearer if written using CLAMP.
Also, please define WORKER_NUM_MIN.
```C
   arg_children_max = CLAMP(MIN(cpu_limit, mem_limit), WORKER_NUM_MIN, WORKER_NUM_MAX)
```
?

Reply:
Strictly speaking, it's not equivalent because the limit of 10 comes with the memory limit. So if the cpu limit wants a lower value, we will follow it.

Reply:
@keszybz does that make sense ?

Reply:
Yeah. OK, let's merge as is.